src/lambda/addLocation.js

```diff
@@ -0,0 +1,36 @@
+import connectToDatabase from './database/dbConnection';
+import User from './database/models/User';
+import { getSingleGeo } from './utils/postcodes';
+import { addRentalRecord } from './database/queries/rentalRecord';
+
+// Stub - function to be replaced with one that gets ID of logged in user
+const getCurrentUserId = async () => {
+  const user = await User.findOne({ name: 'Michael Watts' });
+  return user._id;
+};
+
+export async function handler(event, context) {
+  // eslint-disable-next-line no-param-reassign
+  context.callbackWaitsForEmptyEventLoop = false;
+  if (event.httpMethod === 'POST') {
+    const location = JSON.parse(event.body);
+    try {
+      await connectToDatabase();
+      location.submittedBy = await getCurrentUserId();
+      location.geoLocation = await getSingleGeo(location);
```

Comment:
@arrested-developer  where the location comes from? rental form?

Reply:
I think this should be changed to be like this
```suggestion
      location.geoLocation = await getSingleGeo(location.postcode);
```